walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -56,4 +54,85 @@
 
     return validTimes;
   }
+
+  /** 
+    * Returns a collection of times that work for an optimal number of optional 
+    * people, as well as all the mandatory people.  
+    */
+  private Collection<TimeRange> getOptimalSubsetTimes(
+    Set<String> mandatoryAttendees, Set<String> optionalAttendees, 
+    Collection<Event> events, MeetingRequest request) {
+    
+    Set<Set<String>> subsets = Sets.powerSet(optionalAttendees);
+    Set<Set<String>> infeasibleGroups = new HashSet<Set<String>>();
+    Collection<TimeRange> optimalListOfTimes = new ArrayList<TimeRange>();
+
+    for (int i = 1; i < optionalAttendees.size(); i++) {
```

Comment:
Is using a for loop to iterate over values of `i` the best way to do this?

Reply:
Check all subsets is one option. But seems pretty time consuming, and the time needed grows exponentially with number of people.
Another option is to check all possible time slots and see which time slot can have the most attendees. The time complexity for this one would be N*M, assuming there are N time slots and M attendees.

Reply:
This is a great idea!!! I'm going to try to implement this ASAP and recommit to this PR